awx/ui_next/src/components/CodeMirrorInput/VariablesField.jsx

```diff
@@ -1,65 +1,58 @@
 import React, { useState } from 'react';
 import { string, bool } from 'prop-types';
-import { Field, useFormikContext } from 'formik';
+import { useField } from 'formik';
 import { Split, SplitItem } from '@patternfly/react-core';
 import { yamlToJson, jsonToYaml, isJson } from '@util/yaml';
+import { FormFullWidthLayout } from '@components/FormLayout';
 import CodeMirrorInput from './CodeMirrorInput';
 import YamlJsonToggle from './YamlJsonToggle';
 import { JSON_MODE, YAML_MODE } from './constants';
 
 function VariablesField({ id, name, label, readOnly }) {
-  const { values } = useFormikContext();
-  const value = values[name];
-  const [mode, setMode] = useState(isJson(value) ? JSON_MODE : YAML_MODE);
+  const [field, meta, helpers] = useField(name);
+  const [mode, setMode] = useState(isJson(field.value) ? JSON_MODE : YAML_MODE);
 
   return (
-    <Field name={name}>
-      {({ field, form }) => (
-        <div className="pf-c-form__group">
-          <Split gutter="sm">
-            <SplitItem>
-              <label htmlFor={id} className="pf-c-form__label">
-                <span className="pf-c-form__label-text">{label}</span>
-              </label>
-            </SplitItem>
-            <SplitItem>
-              <YamlJsonToggle
-                mode={mode}
-                onChange={newMode => {
-                  try {
-                    const newVal =
-                      newMode === YAML_MODE
-                        ? jsonToYaml(field.value)
-                        : yamlToJson(field.value);
-                    form.setFieldValue(name, newVal);
-                    setMode(newMode);
-                  } catch (err) {
-                    form.setFieldError(name, err.message);
-                  }
-                }}
-              />
-            </SplitItem>
-          </Split>
-          <CodeMirrorInput
+    <FormFullWidthLayout>
```

Comment:
I'm a little hesitant to put the layout inside this component... It does restrict where/how we can use it.

I guess it's safe to assume this will always be a full width field? Will this FormFullWidthLayout work if the containing form doesn't use our layout components?

Reply:
That's a good point, we can do it at the Form level too